### PR TITLE
Updated Select documentation.

### PR DESCRIPTION
**Pull request by magi42**

### CODE REVIEW

File: frontend/demo/component/select/select-rich.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import { html, LitElement, customElement } from 'lit-element';
+import { render } from 'lit-html';
+import '@vaadin/vaadin-select/vaadin-select';
+import '@vaadin/vaadin-list-box/vaadin-list-box';
+import '@vaadin/vaadin-item/vaadin-item';
+
+@customElement('select-rich')
+export class Example extends LitElement {
+  render() {
+    let people: Array<Array<string>>;
```
[the remark this answers is not in the document]
Comment (jouni) [reply frontend/demo/component/select/select-rich.ts]: No longer applicable, now using the shared data model (Person) instead.

File: src/main/java/com/vaadin/demo/component/select/SelectSeparators.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import com.vaadin.flow.component.html.Div;
+import com.vaadin.flow.component.html.Hr;
+import com.vaadin.flow.component.select.Select;
+import com.vaadin.flow.router.Route;
+
+@Route("select-separators")
+public class SelectSeparators extends Div {
+
+  public SelectSeparators() {
+    class Ordering {
```
[the remark this answers is not in the document]
Comment (jouni) [reply src/main/java/com/vaadin/demo/component/select/SelectSeparators.java:16]: Are we using Ordering in any other example yet? If not, I’m not going to extract it yet.

File: src/main/java/com/vaadin/demo/component/select/SelectRich.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        FlexLayout wrapper = new FlexLayout();
+        
+        Image image = new Image();
+        image.setSrc("https://randomuser.me/api/portraits/women/" +
+                     person.getId() + ".jpg");
+        image.setWidth("42px");
+        
+        Div info = new Div();
+        info.setText(person.getName());
+        info.getStyle().set("margin-left", "10px");
```
[the remark this answers is not in the document]
Comment (jouni) [reply src/main/java/com/vaadin/demo/component/select/SelectRich.java]: Added comments (and cleaned up the styles).

File: src/main/java/com/vaadin/demo/component/select/SelectRich.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+import com.vaadin.demo.DemoExporter; // hidden-full-source-line
+import com.vaadin.flow.component.html.Div;
+import com.vaadin.flow.component.html.Image;
+import com.vaadin.flow.component.orderedlayout.FlexLayout;
+import com.vaadin.flow.component.select.Select;
+import com.vaadin.flow.data.renderer.ComponentRenderer;
+import com.vaadin.flow.router.Route;
+
+@Route("select-rich")
```
[the remark this answers is not in the document]
Comment (jouni) [reply src/main/java/com/vaadin/demo/component/select/SelectRich.java]: Renamed to `select-presentation.ts` and `SelectPresentation.java`

File: src/main/java/com/vaadin/demo/component/select/SelectRich.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+public class SelectRich extends Div {
+
+	public SelectRich() {
+	
+	class Person {
```
[the remark this answers is not in the document]
Comment (jouni) [reply src/main/java/com/vaadin/demo/component/select/SelectRich.java]: Using the shared data model now.

File: frontend/demo/component/select/select-presentation.ts [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+                    <div
+                      style="font-size: var(--lumo-font-size-s); color: var(--lumo-secondary-text-color);">
+                      ${person.profession}
+                    </div>
+                  </div>
+                </vaadin-item>
+                `)}
+            `,
+            root
+          )}
```
Comment (jouni) [inline frontend/demo/component/select/select-presentation.ts:54]: @tomivirkki, this isn’t working for me. No items are rendered.

File: src/main/java/com/vaadin/demo/component/select/SelectSeparators.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (jouni) [reply src/main/java/com/vaadin/demo/component/select/SelectSeparators.java:16]: Oh, now I see. The same model is also defined in `SelectSeparators.java`

[the remark this answers is not in the document]
Comment (jouni) [reply src/main/java/com/vaadin/demo/component/select/SelectSeparators.java]: I created a new `Ordering.java` class which the examples use. Not sure if that’s the way to go, or do we need a similar thing in the TypeScript examples as well?

File: frontend/demo/component/select/select-presentation.ts (hunk 6 above)
Comment (tomivirkki) [reply frontend/demo/component/select/select-presentation.ts]: Fixed in https://github.com/vaadin/docs/pull/21/commits/793b220a2e12a669e16cfafb3b915adaf4b9d627 vaadin-select desperately needs a new post-Polymer API

File: src/main/java/com/vaadin/demo/component/select/SelectSeparators.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (tomivirkki) [reply src/main/java/com/vaadin/demo/component/select/SelectSeparators.java]: Would `Select<String>` work here so a new domain object class (used in just these two examples) wouldn't be needed?

**Comment by tomivirkki** [comment]
TS-examples: We should probably encourage defining a `value` for each `<vaadin-item>`. Without it, the `value` of the `vaadin-select` becomes the text content of the selected item. And that might be something really random (including line breaks or localized content) depending on what gets rendered inside of the item.

**Comment by jouni** [comment]
Replaced by #44

**Pull request closed without merging**